tools/mac-distribution/notarize/submit.go

```diff
@@ -0,0 +1,81 @@
+package notarize
+
+import (
+	"encoding/json"
+
+	"github.com/gravitational/trace"
+)
+
+// SubmissionResponseData contains information about the status of a submission.
+//
+// Reference: https://developer.apple.com/documentation/notaryapi/submissionresponse/data-data.dictionary
+type SubmissionResponseData struct {
+	ID string `json:"id,omitempty"`
+}
+
+// SubmitAndWait is a convenience function that wraps the Submit and WaitForSubmission functions.
+func (t *Tool) SubmitAndWait(pathToPackage string) error {
+	submissionID, err := t.Submit(pathToPackage)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	if err := t.WaitForSubmission(submissionID); err != nil {
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+// Submit will submit a package for notarization.
+// A package must be in the format of a zip, package installer (.pkg), or a disk image (.dmg)
+// A success will return a submission ID which can be polled on later.
+func (t *Tool) Submit(pathToPackage string) (id string, err error) {
+	args := []string{
+		"notarytool",
+		"submit", pathToPackage,
+		"--team-id", t.Creds.BundleID,
+		"--apple-id", t.Creds.AppleUsername,
+		"--apple-password", t.Creds.ApplePassword,
+		"--output-format", "json",
+	}
+
+	stdout, err := t.cmdRunner.RunCommand("xcrun", args...)
```

Comment:
Like mentioned above, in dry run mode this looks like it will dump all arguments, including password, into the logs.